engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/types/EspTypeUtils.scala

```diff
@@ -199,6 +193,13 @@ object EspTypeUtils {
       Typed.genericTypeClass(rawType, paramsType.getActualTypeArguments.toList.flatMap(extractClass))
     } else if (classOf[scala.collection.Iterable[_]].isAssignableFrom(rawType)) {
       Typed.genericTypeClass(rawType, paramsType.getActualTypeArguments.toList.flatMap(extractClass))
+    // All below are for AbstractMethodDefinitionExtractor purpose
```

Comment:
Can we refactor it a bit? I think for all types except StateT we do Typed.genericTypeClass(rawType, paramsType.getActualTypeArguments.toList.flatMap(extractClass))?

Reply:
Yep, I thought that It was explicitly specified because of some kind of optimization? But even if so, it may be true that it was premature optimalization... I will change it but in last PR: https://github.com/TouK/nussknacker/pull/918 to avoid fight with rebasing... Good catch!